src/renderer/utils/storageHelper.ts

```diff
@@ -61,9 +67,8 @@ export class StorageHelper<T> implements StorageLayer<T> {
 
     this.storage = storage;
 
-    // TODO: This code uses undocumented MobX internal to criminally permit exotic mutations without encapsulation.
-    this.data.observe_(({ newValue, oldValue }) => {
-      this.onChange(newValue as T, oldValue as T);
+    observe(this.data, (change) => {
+      this.onChange(change.newValue as T | undefined, change.oldValue as T | undefined);
```

Comment:
This added `| undefined` everywhere seems pretty damaging, yes?

Reply:
Ah no... That is the whole point. To be explicit about types.

I did remove the unnecessary `| undefined` when used in optional fields.

Reply:
The pain is not knowing if something is `null` or `undefined`. It makes it much easier to know when to do defensive code.

In this case the typings from `mobx` are both `unknown` (BTW I don't know why they are `unknown`) so this cast is necessary.

Reply:
Ok. Resolved.